examples/social/src/lib.rs

```diff
@@ -251,33 +250,56 @@ pub struct Key {
 
 // Serialize keys so that the lexicographic order of the serialized keys corresponds to reverse
 // chronological order, then sorted by author, then by descending index.
-impl CustomSerialize for Key {
-    fn to_custom_bytes(&self) -> Result<Vec<u8>, ViewError> {
+impl Serialize for Key {
+    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
         let data = (
             (!self.timestamp.micros()).to_be_bytes(),
             &self.author,
             (!self.index).to_be_bytes(),
         );
-        Ok(bcs::to_bytes(&data)?)
+        let key = bcs::to_bytes(&data).unwrap();
+        serializer.serialize_bytes(&key)
     }
+}
+
+impl<'de1> Deserialize<'de1> for Key {
+    fn deserialize<D: serde::Deserializer<'de1>>(deserializer: D) -> Result<Self, D::Error> {
+        struct Visitor;
+        impl<'de2> serde::de::Visitor<'de2> for Visitor {
+            type Value = Key;
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                write!(formatter, "a serialized key encoded")
+            }
+
+            fn visit_bytes<E: serde::de::Error>(self, v: &[u8]) -> Result<Key, E> {
+                let (time_bytes, author, idx_bytes) =
+                    (bcs::from_bytes(v).map_err(|e| E::custom(format!("{:?}", e))))?;
+                Ok(Key {
+                    timestamp: Timestamp::from(!u64::from_be_bytes(time_bytes)),
+                    author,
+                    index: !u64::from_be_bytes(idx_bytes),
+                })
```

Comment:
Couldn't this just be the body of `deserialize` itself? Then we wouldn't need a `Visitor` impl at all.

```suggestion
                let (time_bytes, author, idx_bytes) =
                    ([u8; 8], ChainId, [u8; 8])::deserialize(deserializer)?;
                Ok(Key {
                    timestamp: Timestamp::from(!u64::from_be_bytes(time_bytes)),
                    author,
                    index: !u64::from_be_bytes(idx_bytes),
                })
```